aten/src/ATen/native/mps/operations/BinaryOps.mm

```diff
@@ -72,16 +72,39 @@ void binaryOpTensor(const Tensor& self, const Tensor& other, const Scalar& alpha
 
           // this type inference is only required at the time of graph creation
           const ScalarType common_dtype = c10::promoteTypes(self.scalar_type(), other.scalar_type());
-          if (self.scalar_type() != common_dtype) {
-            primaryCastTensor = castMPSTensor(mpsGraph, newCachedGraph->primaryTensor, common_dtype);
+
+          // Condition - 
+          // 1. Division operation
+          // 2. Inputs are not float
+          bool div_condition = op_name.rfind("div", 0) == 0
```

Comment:
This div-specific code shouldn't be here in a generic function as it affects other ops and looks hacky with op_name checking. It should be moved to div_mode_template(). You could run the generic binaryOpTensor() conditionally in div_mode_template(), like this:
```
void div_mode_template(...) {
...
if (whatever_condition) {
 // your new custom code
} else {
  // call the generic function like we used to.
  binaryOpTensor(self, other, Scalar(1.0), output, op_name,...);
}
```

Reply:
That's a very good point, I did this because this change has to be made inside the graph creation. If I make another custom code similar to binaryOpTensor(), it will duplicate a lot of code related to graph creation, input checking, placeholder creation. If that's OK I can go ahead with that, what I've currently done involves minimal changes, but like you said it has the op name checking.

Reply:
Please check test_mps, rest of the consistency binary ops, torchbench networks.

Reply:
Yes, I checked all these and it works with the changes